drivers/tee/optee/optee.c

```diff
@@ -874,15 +1007,93 @@ static int optee_shm_unregister(const struct device *dev, struct tee_shm *shm)
 	return rc;
 }
 
-static int optee_suppl_recv(const struct device *dev, uint32_t func, unsigned int num_params,
+static int optee_suppl_recv(const struct device *dev, uint32_t *func, unsigned int *num_params,
 			    struct tee_param *param)
 {
+	struct optee_driver_data *data = (struct optee_driver_data *)dev->data;
+	struct optee_supp *supp = &data->supp;
+	struct optee_supp_req *req = NULL;
+
+	if (supp->current) {
```

Comment:
looks like, it isn't safe
what if somebody sets `supp->current` right after this check from another thread?